controllers/resource_controller.go

```diff
@@ -70,6 +70,17 @@ func (r *ResourceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, retErr
 	}
 
+	snapshot, err := CheckIfSnapshotExists(ctx, r.Client, obj.Spec.SnapshotTemplate.Name, obj.Namespace)
```

Comment:
could also just return a `bool` rather than the entire snapshot?

Reply:
Yep, I actually rewrote that now to do that. :D